Tribler/Policies/credit_mining_util.py

```diff
@@ -0,0 +1,217 @@
+"""
+File containing function used in credit mining module.
+"""
+
+
+import os
+from binascii import hexlify, unhexlify
+
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.simpledefs import NTFY_CHANNELCAST
+from Tribler.Core.simpledefs import NTFY_TORRENTS
+from Tribler.Core.simpledefs import NTFY_VOTECAST
+from Tribler.Main.Utility.GuiDBTuples import CollectedTorrent, RemoteTorrent, NotCollectedTorrent, Channel, \
+    ChannelTorrent
+from Tribler.dispersy.taskmanager import TaskManager
+
+
+def validate_source(source):
+    """
+    Function to check whether a source is a valid source or not
```

Comment:
I mean valid by string literal. As the source can only dispersy id, url, and path directory. Random string won't do.